### ISSUE TITLE
Docs: Mention preference for small, self-contained rule test cases

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6887]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

In general, in our `packages/eslint-plugin/tests/rules/**/*.test.ts` tests, I'm under the impression we generally ask for test cases that are single-purpose.

Good:

```ts
{
  code: `willCauseComplaint(1);`
  errors: [
    { /* ... */ }
  ]
}
```

Not so good, should be split up:

```ts
{
  code: `
    willCauseComplaint(1);
    willCauseComplaint(2);
  `
  errors: [
    { /* ... */ },
    { /* ... */ }
  ]
}
```

I see the latter form come up sometimes in PRs. Let's document our preference for the former?

### Affected URL(s)

https://typescript-eslint.io/maintenance/pull-requests

Or perhaps we'd want to split out a page dedicated to good practices for ESLint rule PRs? Since they're such a large percentage of our PRs & have specific practices associated.

Aside: if this is accepted by other maintainers we should file an issue to fix up old test files that violate this preference. E.g. `no-floating-promises.test.ts`.
